File: 110405/32.c
#include <stdio.h>
#define MAX 1000

static int n, ti[MAX], si[MAX];
static int result[MAX];

int main(void){
	int i, j, t;
	
	scanf("%d", &t);
	for(i = 0; i < t; i++){
		input();
		solve();
		if(i > 0)
			printf("\n");
		for(j = 0; j < n-1; j++)
			printf("%d ", result[j] + 1);
		printf("%d\n", result[n-1] + 1);
	}
}
void input(void){
	int i;
	scanf("%d", &n);
	for(i = 0; i < n; i++)
		scanf("%d %d", &ti[i], &si[i]);
}
void solve(void){
	int i, j, temp;
	for(i = 0; i < n; i++)
		result[i] = i;
	
	for(i = 1; i < n; i++){
		for(j = 0; j < n-i; j++){
			if(ti[result[j]]*si[result[j+1]] > ti[result[j+1]]*si[result[j]]){
				temp = result[j];
				result[j] = result[j+1];
				result[j+1] = temp;
			}
		}
	}
}

File: 110405/32.java
import java.io.*;
import java.util.Scanner;
import java.util.HashMap;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;
import java.util.Collections;
	
class Main {
	public static void main(String[] args) throws Exception {
		Scanner input = new Scanner(System.in);
		int n = Integer.parseInt(input.nextLine());
		for(int i =0; i<n; i++) {
			String garbage = input.nextLine();
			int cnt = Integer.parseInt(input.nextLine());
			HashMap<Integer,Double> result = new HashMap<>();
			
			for(int h =1; h<cnt+1; h++) {
				String[] temp = input.nextLine().split(" ");
				double res = (Double.parseDouble(temp[1])/Double.parseDouble(temp[0])); 
				result.put(h,res);
			}
	
			List<Integer> list = new ArrayList<>(result.keySet());
			Collections.sort(list, (value1, value2) -> (result.get(value2).compareTo(result.get(value1))));
			for(int key : list) {
				System.out.print(key + " ");
			}
			System.out.println("");
			System.out.println("");
		}
		/**S1/T1 -> 다 구한뒤에 오름차순으로 정렬 퀵소트로 구현 가능하면 퀵 -> 안되면 걍 삽입**/
	}
}